packages/headless/src/api/knowledge/createAnswerAPISlice.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {ConfigurationSection} from '../../state/state-sections';
+import {createAbortController} from '../../utils/abort-controller-polyfill';
+import {baseKnowledgeAPISlice} from './baseKnowledgeAPISlice';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+const abortController = createAbortController();
```

Comment:
Having this a global module variable is problematic.

There is an inherent statefulness related to aborting request, which you don't want to handle at a global level.

Also, I'm not sure I understand how the abort controller is used (if at all?) by RTK ? 

Where does this `abortController` gets passed to the `Fetch` request ? I see we reference it `onError`, but for it to do anything, it must be passed to the underlying fetch.

I have a hunch it does not do anything right now.

Reply:
I removed the polyfilled implementation of the `abortController` since the MVP of the new API flow will not be used in Quantic/Saleforce where it might be required. We'll cross the bridge when we get there.